src/rules/prefer-in-document.js

```diff
@@ -201,16 +270,25 @@ export const create = (context) => {
     [`MemberExpression[object.callee.name=expect][property.name=${alternativeMatchers}][object.arguments.0.type=Identifier]`](
       node
     ) {
-      const queryNode = getAssignmentForIdentifier(
+      // Value expression being assigned to the left-hand value
+      const rightValueNode = getAssignmentForIdentifier(
         context,
         node.object.arguments[0].name
       );
+
+      // Not a DTL query
+      if (!rightValueNode || rightValueNode.type !== "CallExpression") {
```

Comment:
I think we can use optional chaining here:

```suggestion
      if (rightValueNode?.type !== "CallExpression") {
```